tests/bootstrap/test_cli.py

```diff
@@ -178,3 +178,21 @@ def test_container_starts_non_root(self, runner, monkeypatch, container_client):
 
         output = container_client.exec_in_container(config.MAIN_CONTAINER_NAME, ["ps", "-u", user])
         assert "supervisord" in to_str(output[0])
+
+    # TODO: enable test in next iteration, after a full rebuild of the Docker image
+    @pytest.mark.skip
+    def test_start_cli_within_container(self, runner, container_client):
+        output = container_client.run_container(
+            "localstack/localstack",
+            remove=True,
+            entrypoint="",
+            command=["bin/localstack", "start", "-d"],
+            mount_volumes=[("/var/run/docker.sock", "/var/run/docker.sock")],
+            env_vars={"LOCALSTACK_VOLUME_DIR": "/tmp/ls-volume"},
+        )
+        stdout = to_str(output[0])
+        assert "starting LocalStack" in stdout
```

Comment:
I'm not sure it's a good idea to make this test dependent on the current latest docker image. Basically if at any point this test fails in CI because of a broken image we'll have to manually intervene since they block each other (test doesn't work because the image is broken and no new latest image is released because the CI tests fail).

Reply:
I'd suggest we build some image locally here that uses docker to run some other container (e.g. echo via busybox) and use that to verify the change here :thinking:

Reply:
Good point, would definitely be beneficial to have this included in this PR directly, without needing to rebuild `latest`. I'll try a few things - perhaps we can mount the local code into the container 👍  (I'd like to avoid doing a full image build, not to blow up the CI time for a relatively small feature).

Reply:
Seems to work reasonably well with the volume mount 👌  - updated the PR